File: demo03_ws/plumbing_pub_sub/src/demo04_sub_person.cpp
#include "ros/ros.h"
#include "plumbing_pub_sub/Person.h"

/* 
    订阅人的消息
        1.包含头文件
            #include "plumbing_pub_sub/Person.h"
        2.初始化ros节点
        3.创建ros节点句柄
        4.创建订阅者对象
        5.编写回调函数，处理订阅的逻辑
        6.调用spin（）函数
 */

void doPerson(const plumbing_pub_sub::Person::ConstPtr& person){
    ROS_INFO("订阅的人的信息：%s,%d,%.2f",person->name.c_str(),person->age,person->height);
}

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
    ROS_INFO("订阅方实现");
        // 2.初始化ros节点
    ros::init(argc,argv,"jiazhang");
        // 3.创建ros节点句柄
    ros::NodeHandle nh;
        // 4.创建订阅者对象
    ros::Subscriber sub=nh.subscribe("liaotian",10,doPerson);
        // 5.编写回调函数，处理订阅的逻辑
        // 6.调用spin（）函数
    ros::spin();
    return 0;
}

File: demo03_ws/plumbing_pub_sub/scripts/demo04_sub_person.py
#! /usr/bin/env python
import rospy 
from plumbing_pub_sub.msg import Person
""" 
订阅方：订阅人的消息数据
    1.导包
    2.初始化ros节点
    3.创建订阅者对象
    4.处理订阅到的数据，回调函数
    5.spin()
 """
def doPerson(p):
    rospy.loginfo("订阅到的数据：%s,%d,%.2f",p.name,p.age,p.height)

if __name__=="__main__":
    # 2.初始化ros节点
    rospy.init_node("daye")
    # 3.创建订阅者对象
    sub=rospy.Subscriber("jiaoshetou",Person,doPerson)
    # 4.处理订阅到的数据，回调函数
    # 5.spin()
    rospy.spin()